### PR TITLE
Add Chainguard OIDC provider.

### PR DESCRIPTION
**Pull request by mattmoor**
#### Summary

This adds support for Chainguard issued tokens, so that users can sign with their Chainguard-issued identity, and so that we can explore signing our own content with our internal service principal construct (see issue).

Related: https://github.com/sigstore/fulcio/issues/1702

#### Release Note

Add support for signing with Chainguard-issued tokens.

#### Documentation

This is mostly intended for our own use, but I'd be happy to document how folks can make use of this with `chainctl auth token --audience sigstore` if someone points me in the right direction.

### CODE REVIEW

**Comment by haydentherapper** [comment]
Is this a provider you would want enabled in the public instance as well? If so, do you have an estimated QPS we should expect?

File: go.mod
Comment (haydentherapper) [inline go.mod:3]: Let's split this out into its own PR, cc @cpanato

**Comment by mattmoor** [comment]
> Is this a provider you would want enabled in the public instance as well? If so, do you have an estimated QPS we should expect?

Yes.  No real estimation, but I would expect that if anything this would cannibalize QPS we are currently sending with github actions tokens.  I could work with someone on the rotation to try and figure out what that might look like, if it's important.

File: go.mod
Comment (mattmoor) [reply go.mod:3]: https://github.com/sigstore/fulcio/pull/1706

**Comment by mattmoor** [comment]
Rebased on @cpanato go change

**Comment by mattmoor** [comment]
... and fixed merge conflict that drops `CompactSerialize()` method.

**Review by haydentherapper — APPROVED** [review]
LGTM - Do you want to hold off until you've added the custom OIDs, or merge this as is?

**Comment by mattmoor** [comment]
I'd say we land this as-is and we can make additive changes as the need for enriching things under our own OID arises. 👍

**Review by priyawadhwa — APPROVED** [review]

**Review by cpanato — COMMENTED** [review]
late lgtm

**Comment by mattmoor** [comment]
Let me know if there are instructions to test against staging (and when I can expect that to work!) 🙏

**Comment by haydentherapper** [comment]
I queued up the change in staging, I'll update here once it's rolled out. https://docs.sigstore.dev/system_config/public_deployment/#usage has instructions on how to test.

**Comment by mattmoor** [comment]
@haydentherapper this seems out of date:
```
curl -O https://raw.githubusercontent.com/sigstore/root-signing/main/staging/repository/1.root.json
```

I poked around in the repo, and I don't see an obvious replacement for the staging root.

**Comment by haydentherapper** [comment]
https://github.com/sigstore/docs/pull/315 for the doc fix

**Comment by mattmoor** [comment]
If I use a token without an embedded email claim:
```
# cosign sign --oidc-issuer "https://oauth2.sigstage.dev/auth" --fulcio-url "https://fulcio.sigstage.dev" --rekor-url "https://rekor.sigstage.dev" ttl.sh/mattmoor/static --identity-token "$(chainctl auth token --audience sigstore)"
Generating ephemeral keys...
Retrieving signed certificate...
Successfully verified SCT...
WARNING: Image reference ttl.sh/mattmoor/static uses a tag, not a digest, to identify the image to sign.
    This can lead you to sign a different image than the intended one. Please use a
    digest (example.com/ubuntu@sha256:abc123...) rather than tag
    (example.com/ubuntu:latest) for the input to cosign. The ability to refer to
    images by tag will be removed in a future release.


        The sigstore service, hosted by sigstore a Series of LF Projects, LLC, is provided pursuant to the Hosted Project Tools Terms of Use, available at https://lfprojects.org/policies/hosted-project-tools-terms-of-use/.
        Note that if your submission includes personal data associated with this signed artifact, it will be part of an immutable record.
        This may include the email address associated with the account with which you authenticate your contractual Agreement.
        This information will be used for signing this artifact and will be stored in public transparency logs and cannot be removed later, and is subject to the Immutable Record notice at https://lfprojects.org/policies/hosted-project-tools-immutable-records/.

By typing 'y', you attest that (1) you are not submitting the personal data of any other person; and (2) you understand and agree to the statement and the Agreement terms at the URLs listed above.
Are you sure you would like to continue? [y/N] y
tlog entry created with index: 29985605
Pushing signature to: ttl.sh/mattmoor/static

# cosign verify --rekor-url "https://rekor.sigstage.dev" --certificate-identity=https://issuer.enforce.dev/720909c9f5279097d847ad02a2f24ba8f59de36a/a033a6fabe0bfa0d --certificate-oidc-issuer=https://issuer.enforce.dev ttl.sh/mattmoor/static 

Verification for ttl.sh/mattmoor/static:latest --
The following checks were performed on each of these signatures:
  - The cosign claims were validated
  - Existence of the claims in the transparency log was verified offline
  - The code-signing certificate was verified using trusted certificate authority certificates

[{"critical":{"identity":{"docker-reference":"ttl.sh/mattmoor/static"},"image":{"docker-manifest-digest":"sha256:e78eb21f59f52446a23f7d45e78805c4f26406a6e9c4b21ec50dfdf07e6bec57"},"type":"cosign container image signature"},"optional":{"1.3.6.1.4.1.57264.1.1":"https://issuer.enforce.dev","Bundle":{"SignedEntryTimestamp":"MEQCIEuot42CKdxsUA42w2pIBNgvyK0zoUf8s5loJ6JEZQIPAiBV1MMN6FTginw5xJUEehZNj9meIFp+eVyzLJJQ1A/L/A==","Payload":{"body":"eyJhcGlWZXJzaW9uIjoiMC4wLjEiLCJraW5kIjoiaGFzaGVkcmVrb3JkIiwic3BlYyI6eyJkYXRhIjp7Imhhc2giOnsiYWxnb3JpdGhtIjoic2hhMjU2IiwidmFsdWUiOiJlOWE2MWFhZjE3YmYyZmI1NjM2M2U5YmU0ZGQzMjk0MWE2MzE3OTBjZDVjNzZlNDJlZWU2ZWM2OGMzZmZiNjQ2In19LCJzaWduYXR1cmUiOnsiY29udGVudCI6Ik1FVUNJUURIaWt3VlkrQ0RJeGEzNXg2Y3FmTi9RL0xEdkNhcU9JN0F2TlY1d0c0WmtRSWdlTGFxaTM5amMwSFplZXIyQitYcFM0OE1tNFhCaHNnZVczNjBIUHF5a05nPSIsInB1YmxpY0tleSI6eyJjb250ZW50IjoiTFMwdExTMUNSVWRKVGlCRFJWSlVTVVpKUTBGVVJTMHRMUzB0Q2sxSlNVUkVWRU5EUVhCTFowRjNTVUpCWjBsVlNUQTVRMnN6V0hSWmNFVTBZWEl6YjJSWGQyWm1kRkJUY25jNGQwTm5XVWxMYjFwSmVtb3dSVUYzVFhjS1RucEZWazFDVFVkQk1WVkZRMmhOVFdNeWJHNWpNMUoyWTIxVmRWcEhWakpOVWpSM1NFRlpSRlpSVVVSRmVGWjZZVmRrZW1SSE9YbGFVekZ3WW01U2JBcGpiVEZzV2tkc2FHUkhWWGRJYUdOT1RXcFJkMDU2UlhkTlZHZDVUVVJKTkZkb1kwNU5hbEYzVG5wRmQwMVVaM3BOUkVrMFYycEJRVTFHYTNkRmQxbElDa3R2V2tsNmFqQkRRVkZaU1V0dldrbDZhakJFUVZGalJGRm5RVVY0VldkNWRFc3ZTbk5TVTFKNWVXUXhjU3RIVXpKeFVFTnhTRWMyWWxKUU1XNUthRVVLTTFWQ1RuZHdSREJoVFdKMWQzcGhRWE14TWtadVJFMXhjMjVEVkVZMGVrMUdkV2h5TkM4eGN6aFRNbFY2TlVob1kzRlBRMEZpUlhkblowZDBUVUUwUndwQk1WVmtSSGRGUWk5M1VVVkJkMGxJWjBSQlZFSm5UbFpJVTFWRlJFUkJTMEpuWjNKQ1owVkdRbEZqUkVGNlFXUkNaMDVXU0ZFMFJVWm5VVlZqSzJOR0NtczNPRnB3U0hKd2NIaGpTemgyTTBSbFdGRkVNVGRCZDBoM1dVUldVakJxUWtKbmQwWnZRVlZqV1ZsM2NHaFNPRmx0THpVNU9XSXdRbEp3TDFndkwzSUtZalozZDFsbldVUldVakJTUVZGSUwwSkdaM2RXYjFwVllVaFNNR05JVFRaTWVUbHdZek5PTVZwWVNYVmFWelZ0WWpOS2FscFROV3RhV0ZsMlRucEpkd3BQVkVFMVdYcHNiVTVVU1ROUFZFRTFUakpSTkU1RVpHaGFSRUY1V1ZSS2JVMXFVbWxaVkdodFRsUnNhMXBVVFRKWlV6bG9UVVJOZWxsVVdtMVpWMHBzQ2sxSFNtMVpWRUpyVFVOblIwTnBjMGRCVVZGQ1p6YzRkMEZSUlVWSGJXZ3daRWhDZWs5cE9IWmhXRTU2WkZkV2VVeHRWblZhYlRsNVdUSlZkVnBIVmpJS1RVTnZSME5wYzBkQlVWRkNaemM0ZDBGUlowVklRWGRoWVVoU01HTklUVFpNZVRsd1l6Tk9NVnBZU1hWYVZ6VnRZak5LYWxwVE5XdGFXRmwzWjFselJ3cERhWE5IUVZGUlFqRnVhME5DUVVsRlpsRlNOMEZJYTBGa2QwRnlUVXg2WTJGSmFrbzBkVWhaU21sc1pXUkNPVWxQVkVkWFFYWkxZMDA0ZEdWUk1FUXJDbk54ZVVkbFowRkJRVnBEWkRSQ2RIVkJRVUZGUVhkQ1NVMUZXVU5KVVVSSldYRXphVkZ3UVdKbVFsTkhWa3BHVEd4VmRrRnlNVzVJZDI5Tk5IRmtWR0lLVWpaWE5IZElWamxIWjBsb1FVcEpiV3BzV2l0NlRIZFRhWGd3YkVGTFNVWkhRMkozVTBWM2FqTjNUbk5YVGs1cFJFMTZZMnREVkU5TlFXOUhRME54UndwVFRUUTVRa0ZOUkVFeWEwRk5SMWxEVFZGRGRFRkxlRWwzUzFreFVUWnRiek15U0haUFYwRjNiRVpGSzFGblZGaHVkVk5ZY2pGTFJFRlNPQ3RHT0dsd0NuVjJkRmhoU2s1b01IWmFhMVJaWVVObWNFMURUVkZFZG14VldVTmFiVWxZVkUwNVZtTTVhRXMzVTA5R2NEUnhaRkpzZUZOT1MwbHRNM2hLV2tkR1IzQUtUSEZzSzB0aWJuZHRaM0pvU1ZNNGQyeE5aMnBJZEhjOUNpMHRMUzB0UlU1RUlFTkZVbFJKUmtsRFFWUkZMUzB0TFMwSyJ9fX19","integratedTime":1720635631,"logIndex":29985515,"logID":"d32f30a3c32d639c2b762205a21c7bb07788e68283a4ae6f42118723a1bea496"}},"Issuer":"https://issuer.enforce.dev","Subject":"https://issuer.enforce.dev/720909c9f5279097d847ad02a2f24ba8f59de36a/a033a6fabe0bfa0d"}},{"critical":{"identity":{"docker-reference":"ttl.sh/mattmoor/static"},"image":{"docker-manifest-digest":"sha256:e78eb21f59f52446a23f7d45e78805c4f26406a6e9c4b21ec50dfdf07e6bec57"},"type":"cosign container image signature"},"optional":{"1.3.6.1.4.1.57264.1.1":"https://issuer.enforce.dev","Bundle":{"SignedEntryTimestamp":"MEUCIGFV48y76LM8ljQJWzhIuZP3XjoJENxHpn0/7Rcx3kqRAiEAjSYfw7qay1Iw81bQjiubQZHW+eVpVIQcGLCpxsct6sU=","Payload":{"body":"eyJhcGlWZXJzaW9uIjoiMC4wLjEiLCJraW5kIjoiaGFzaGVkcmVrb3JkIiwic3BlYyI6eyJkYXRhIjp7Imhhc2giOnsiYWxnb3JpdGhtIjoic2hhMjU2IiwidmFsdWUiOiJlOWE2MWFhZjE3YmYyZmI1NjM2M2U5YmU0ZGQzMjk0MWE2MzE3OTBjZDVjNzZlNDJlZWU2ZWM2OGMzZmZiNjQ2In19LCJzaWduYXR1cmUiOnsiY29udGVudCI6Ik1FWUNJUURWSWE5K015dUcrTTlkYjRvT0l0d3N2WmdhaGtmL3I3dC9DV0ZvOXJwU2FBSWhBTERaQkxvQW1WVzRGUElnb1RCNTlsY1ZGdG1rcVRDMkF5VUgrekhldkpMUiIsInB1YmxpY0tleSI6eyJjb250ZW50IjoiTFMwdExTMUNSVWRKVGlCRFJWSlVTVVpKUTBGVVJTMHRMUzB0Q2sxSlNVUkVSRU5EUVhCTFowRjNTVUpCWjBsVlpYUkhhbFlyVlRGcmNHSXZNRGxtWTJKeFZUUTJXVk5hWWxOWmQwTm5XVWxMYjFwSmVtb3dSVUYzVFhjS1RucEZWazFDVFVkQk1WVkZRMmhOVFdNeWJHNWpNMUoyWTIxVmRWcEhWakpOVWpSM1NFRlpSRlpSVVVSRmVGWjZZVmRrZW1SSE9YbGFVekZ3WW01U2JBcGpiVEZzV2tkc2FHUkhWWGRJYUdOT1RXcFJkMDU2UlhkTlZHZDVUWHBCZUZkb1kwNU5hbEYzVG5wRmQwMVVaM3BOZWtGNFYycEJRVTFHYTNkRmQxbElDa3R2V2tsNmFqQkRRVkZaU1V0dldrbDZhakJFUVZGalJGRm5RVVZFVFd0Nk1YZGxhVXhsVTJ4cVZrTXZVRkV4WkVOVE9YVlpWM0pvVUdzck4wc3dUa0lLZGsxdGJGZEhSV05UTVdsclIwRTBibTUxVjBzMlYwbEJPSEEyWTNSc1FrMHZNMjloZFhoMk5rOUxXbUZPWjA1Q1FuRlBRMEZpUlhkblowZDBUVUUwUndwQk1WVmtSSGRGUWk5M1VVVkJkMGxJWjBSQlZFSm5UbFpJVTFWRlJFUkJTMEpuWjNKQ1owVkdRbEZqUkVGNlFXUkNaMDVXU0ZFMFJVWm5VVlZsY0RkdENpdHBZMWhuYlVaMVVVTlJUell3VmxaVmIwSlhiVzQwZDBoM1dVUldVakJxUWtKbmQwWnZRVlZqV1ZsM2NHaFNPRmx0THpVNU9XSXdRbEp3TDFndkwzSUtZalozZDFsbldVUldVakJTUVZGSUwwSkdaM2RXYjFwVllVaFNNR05JVFRaTWVUbHdZek5PTVZwWVNYVmFWelZ0WWpOS2FscFROV3RhV0ZsMlRucEpkd3BQVkVFMVdYcHNiVTVVU1ROUFZFRTFUakpSTkU1RVpHaGFSRUY1V1ZSS2JVMXFVbWxaVkdodFRsUnNhMXBVVFRKWlV6bG9UVVJOZWxsVVdtMVpWMHBzQ2sxSFNtMVpWRUpyVFVOblIwTnBjMGRCVVZGQ1p6YzRkMEZSUlVWSGJXZ3daRWhDZWs5cE9IWmhXRTU2WkZkV2VVeHRWblZhYlRsNVdUSlZkVnBIVmpJS1RVTnZSME5wYzBkQlVWRkNaemM0ZDBGUlowVklRWGRoWVVoU01HTklUVFpNZVRsd1l6Tk9NVnBZU1hWYVZ6VnRZak5LYWxwVE5XdGFXRmwzWjFselJ3cERhWE5IUVZGUlFqRnVhME5DUVVsRlpsRlNOMEZJYTBGa2QwRnlUVXg2WTJGSmFrbzBkVWhaU21sc1pXUkNPVWxQVkVkWFFYWkxZMDA0ZEdWUk1FUXJDbk54ZVVkbFowRkJRVnBEWkRSdVFWbEJRVUZGUVhkQ1NVMUZXVU5KVVVRek9YZFZMeXRuUVROMGJtZFdha0ZWUW1OQ1draHNOV2gwTkRodmNEbGxObU1LUzFkTVFVMDNSamhHZDBsb1FWQjJkak40TjNaQlIydzNhVGczVG5GWlVISlZjblpXUTFOWGFtcDBPWGhKUkVsYWN6TXJhM1ZJT1ZCTlFXOUhRME54UndwVFRUUTVRa0ZOUkVFeVowRk5SMVZEVFZGRFExcG1jR0V3VmtOS2FYbHNTbEpDT0RrMlpHMXlUME5XYlZselpXaEdWRE0zWVVOQmJHWnZhMmxhVTJkTkNtTk1SVXMwT1c1eVprVllieTlvZWtGaWFtTkRUVVEwWTFWWlYyRkVPR2R5YTFWa2JVTmhVbUZFWm5sU1kwcEljV2RLTUZGc1REVm5SbUZ2U0hBeWFrTUtOVlJGUm1nNVdsb3lSRlEwVTFOUE5XcEthVWg1UVQwOUNpMHRMUzB0UlU1RUlFTkZVbFJKUmtsRFFWUkZMUzB0TFMwSyJ9fX19","integratedTime":1720635783,"logIndex":29985605,"logID":"d32f30a3c32d639c2b762205a21c7bb07788e68283a4ae6f42118723a1bea496"}},"Issuer":"https://issuer.enforce.dev","Subject":"https://issuer.enforce.dev/720909c9f5279097d847ad02a2f24ba8f59de36a/a033a6fabe0bfa0d"}}]
```

**Pull request merged**
